### ISSUE TITLE
Semantic line feeds issue in nested bullets in numbered lists

### ISSUE THREAD

**Issue by jamesbraza** [KyleKing/mdformat-mkdocs#7]
Same issue as https://github.com/KyleKing/mdformat-mkdocs/issues/4, but for nested bullets inside a list.

1. Line
   semantic line 1 (3 spaces deep)
    - Bullet (4 spaces deep)
      semantic line 2 (6 spaces deep)

I believe `mdformat-mkdocs==1.0.2` is impacting the second semantic line feed.

---

I have `prettier` (`.prettierrc.toml`: `tabWidth = 4`) and `mdformat`, together in `pre-commit`:

```yaml
    - repo: https://github.com/pre-commit/mirrors-prettier
      rev: v3.0.0
      hooks:
          - id: prettier
    - repo: https://github.com/executablebooks/mdformat
      rev: 0.7.16
      hooks:
          - id: mdformat
            additional_dependencies:
                - mdformat-black
                - mdformat-footnote
                - mdformat-frontmatter
                - mdformat-gfm
                - mdformat-mkdocs>=1.0.2rc2
            args: [--align-semantic-breaks-in-numbered-lists]
```

**Comment by KyleKing** [comment]
I think this is fixed by: https://github.com/KyleKing/mdformat-mkdocs/pull/8

Right now the changes only support semantic indents on bulleted items within a numbered list (i.e. numbered-bullet), but not when the numbered list is within a bulleted list (i.e. bullet-number-bullet). Does the latter also need to be supported? It adds a bit of extra complexity

**Comment by jamesbraza** [comment]
I think ultimately the right solution is supporting recursive permutations of any combination of bullets and numbered lists

One other case I just hit: simple bulleted lists.  Can you add a case for that?

**Comment by KyleKing** [comment]
Ah, I didn't think bulleted lists were handled correctly in `mkdocs` with a semantic indent which is why the feature was limited to numbered lists

Instead, it looks like they work fine and I generalized the logic in the PR. The important notes are:

1. The CLI argument will now be `--align-semantic-breaks-in-lists` instead of  `--align-semantic-breaks-in-numbered-lists`
2. All lines are indented to the same semantic indent. Is this desired or should it be multiples of four when there is a new line? Example:

    ```txt
    1. Line
       semantic indent

       *Should this line be indented with 3 or 4 spaces?*
    ```

**Comment by KyleKing** [comment]
I released v1.0.3 while pending input on the above questions: https://github.com/KyleKing/mdformat-mkdocs/issues/7#issuecomment-1641069209

Let me know if v1.0.3 and the new CLI arg work as expected

**Comment by jamesbraza** [comment]
> 2. All lines are indented to the same semantic indent. Is this desired or should it be multiples of four when there is a new line? Example:
>    ```
>    1. Line
>       semantic indent
>    
>       *Should this line be indented with 3 or 4 spaces?*
>    ```

I think the asterisk line should be indented with 3 spaces for vanilla Markdown.  You will want to confirm this with Mkdocs-style Markdown as well.

Also, thank you for the quick turnaround, it's awesome!

**Comment by KyleKing** [comment]
You're welcome; I'm happy to see my projects getting use!

**Comment by KyleKing** [comment]
So `mkdocs` expects 4-space indents, so valid `mkdocs` markdown looks like this:

https://github.com/KyleKing/mdformat-mkdocs/pull/10/files#diff-f21280fed25185b9d724c03bd5cf1397004cc3f24a6537f77d023e35b4655576

I released a preview version (`1.0.4rc1`) if you want to experiment with these changes: https://pypi.org/project/mdformat_mkdocs/1.0.4rc1/

**Comment by jamesbraza** [comment]
Seems like a step forward with a more general arg name, thanks!

Now if only `prettier` would be as responsive as you (https://github.com/prettier/prettier/issues/5019), I could un-ignore all my Markdown files and use this tool so they all coexist.

**Comment by KyleKing** [comment]
I've released v1.0.4! Let me know if you run into any additional edge cases
